src/main/java/me/asleepp/SkriptItemsAdder/other/util/UpdateChecker.java

```diff
@@ -0,0 +1,76 @@
+package me.asleepp.SkriptItemsAdder.other.util;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import me.asleepp.SkriptItemsAdder.SkriptItemsAdder;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+
+public class UpdateChecker implements Listener {
+
+    private final JavaPlugin plugin;
+    private final String currentVersion;
+    private String latestVersion;
+
+    public UpdateChecker(JavaPlugin plugin) {
+        this.plugin = plugin;
+        this.currentVersion = plugin.getDescription().getVersion();
+        Bukkit.getPluginManager().registerEvents(this, plugin);
+        checkForUpdate();
+    }
+
+    @EventHandler
+    public void onPlayerJoin(PlayerJoinEvent event) {
+        Player player = event.getPlayer();
+        if (player.hasPermission("skript-itemsadder.update.check") && latestVersion != null) {
+            if (!currentVersion.equals(latestVersion)) {
+                player.sendMessage(" ");
+                player.sendMessage(MiniMessage.miniMessage().deserialize("<dark_red>[<red>skript-itemsadder<dark_red>] <white>skript-itemsadder is <red><bold>OUTDATED</bold><white>!"));
+                player.sendMessage(MiniMessage.miniMessage().deserialize("<dark_red>[<red>skript-itemsadder<dark_red>] <white>New version: <gold>" + latestVersion));
+                player.sendMessage(MiniMessage.miniMessage().deserialize("<dark_red>[<red>skript-itemsadder<dark_red>] <white>Download <gold><click:open_url:https://github.com/Asleeepp/skript-itemsadder/releases><hover:show_text:'<red>Click here to get the latest version!'>here<white>!</click>"));
```

Comment:
you could use your mini message utility method here. though im pretty sure player has a method like `sendRichMessage` that uses minimessage parsing internally to format the string